### PR TITLE
Update GPU Examples

### PR DESCRIPTION
**Pull request by mrocklin**
Updating a couple things ahead of PyCon DE

- PyTorch Optuna notebook
    - Now uses package sync instead of an explicit software environment
    - Adds optuna-integrations pip package (needed now I guess)
- PyTorch notebook
    - stripped out optuna stuff, just a single coiled.function call (although it's not much faster on this example)

cc @jrbourbeau @fjetter @phofl

### CODE REVIEW

File: pytorch.yml [hunk 1]
```diff
@@ -16,3 +14,6 @@ dependencies:
   - torchvision
   - pynvml
   - matplotlib
+  - pip
+  - pip:
+    - optuna-integration
```
Comment (jrbourbeau) [inline pytorch.yml:19]: Why `pip`? It looks like this is available on conda-forge https://anaconda.org/conda-forge/optuna-integration

**Review by jrbourbeau — APPROVED** [review]
Thanks @mrocklin -- I'll want to do something similar with our existing GPU examples in the docs

File: pytorch.yml (hunk 1 above)
Comment (mrocklin) [reply pytorch.yml:19]: Ah, I tried briefly and mamba didn't find it.  It's worth checking again though.

Comment (jrbourbeau) [reply pytorch.yml:19]: I tried locally with these changes and mamba resolved the environment 

```suggestion
  - optuna-integration
```

**Pull request merged**